src/framework/ui/view/colordialog.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "colordialog.h"
+
+#include <QApplication>
+#include <QKeyEvent>
+#include <QWindow>
+#include <QTimer>
+
+using namespace muse::ui;
+
+//! NOTE Copied from src/framework/uicomponents/view/topleveldialog.cpp
+
+ColorDialog::ColorDialog(QWidget* parent)
+    : QColorDialog(parent), muse::Injectable(muse::iocCtxForQWidget(this))
+{
+    // We want some windows to be on top of the main window.
+    // But not on top of all other applications when MuseScore isn't active.
+    // On Windows, we achieve this by setting the transient parent.
+    // On macOS, we have to use a workaround:
+    // When the application becomes active, the windows will get the StayOnTop hint.
+    // and when the application becomes inactive, the hint will be removed.
+#ifdef Q_OS_MAC
```

Comment:
Probably caused by this section here, which is not necessary for the color dialog because it's a native dialog.